torch/nested/_internal/sdpa.py

```diff
@@ -736,9 +736,30 @@ def jagged_scaled_dot_product_attention(
             attention.squeeze(0), output_nt_info["offsets"]
         ).transpose(1, 2)
     elif backend_choice == SDPBackend.MATH:
-        return torch._scaled_dot_product_attention_math(
+        # convert jagged layout Nested Tensor to strided layout Nested Tensor
+        # which support the math implementation of SDPA
+        def get_strided_layout_nested_tensor(jagged_layout_nt):
+            lengths = jagged_layout_nt._offsets[1:] - jagged_layout_nt._offsets[:-1]
+            transpose = torch.transpose(jagged_layout_nt, 1, 2)
+            tensor_list = buffer_from_jagged(transpose).split(list(lengths), dim=0)
+            c_nt = torch.nested.as_nested_tensor(list(tensor_list))
+            c_nt = c_nt.transpose(1, 2).contiguous()
+            return c_nt
+
+        query = get_strided_layout_nested_tensor(query)
+        key = get_strided_layout_nested_tensor(key)
+        value = get_strided_layout_nested_tensor(value)
+
+        attn_out = torch._scaled_dot_product_attention_math(
             query, key, value, attn_mask, dropout_p, is_causal, scale=scale
         )[0]
+        attn_out = attn_out.transpose(1, 2)
+        attn_out = torch.nested.as_nested_tensor(
+            list(attn_out.unbind()), layout=torch.jagged
+        )
+        attn_out = attn_out.transpose(1, 2)
+
+        return attn_out
```

Comment:
probably better to put this in NestedTensor's `__torch_function__` instead

adding this should be fine to land to unblock, but we'll eventually want to dedupe with https://github.com/pytorch/pytorch/pull/115749

Reply:
Do you mean add the "_scaled_dot_product_attention_math" in jagged_torch_function in ops.py instead? Wondering why it is better to put it there? It seems to add an extra routing?

Reply:
We actually already have a route for sdpa there and the math path, see:  https://github.com/pytorch/pytorch/blob/main/torch/nested/_internal/sdpa.py#L738

Do you know what happens today if we reach there?

Reply:
This is where I am adding the logic, no? Did I miss something?
Today, this path will cause "RuntimeError: scaled_dot_product_attention: If inputs are nested tensors they must be contiguous".

Reply:
No, I just misread lol